### PR TITLE
Fix cocoa bug

### PR DESCRIPTION
**Pull request by menduz**
GLFW documentation explicitly says that .fullscreen windows must not be resizeable, but that's not the case in this particular setup. If the uppermost pixel of the screen is dragged down, the window gets resized. 
- This PR fixes the issue,
- also adds `setResizable(bool)` and `resizable() bool` functions, 
- and fixes the size of the window in case of restoring .windowed mode, to prevent using the entire size of the display.
- fixes a scaling problem for mouse positions
- adds `.pixel_scale` to the `.framebuffer_resized` event, to detect when we need to change the pixel density/dpi/scaling of the application
- and emits one `.framebuffer_resized` event at the beginning of the application

- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,5 @@ zig-out/
 /docgen_tmp/
 
 examples/libs/
+
+.devcontainer
```
Comment (emidoots) [inline .gitignore]: Can you remove this? Whatever it is, it should go in your global `~/.gitignore` file (details on how to add that at the top of this file.)

**Review by emidoots — APPROVED** [review]
Thank you! I left one minor comment, once that is resolved will merge.

Appreciate you sending this!

**Comment by menduz** [comment]
@slimsag I've pushed other fixes. I'm testing on mac+windows using combinations of multiple DPI+fullscreen/windowed on each device

**Comment by emidoots** [comment]
It looks like you've tried to make mach-core agnostic of pixel densities with the new changes, I'm not necessarily opposed to that I will need to sit down and really read through these changes more closely as I am not sure this is the right choice and this will have impacts for all existing mach-core users.

**Comment by menduz** [comment]
Makes total sense, I'd be happy to evaluate more fine-grained options to prevent breaking the default behavior. As-is, this may be a breaking change, but at high pixel density (as it is right now) makes every application barely usable. Adopting a best-case scenario and making it consistent sounded like a good approach, it is at the end of the day what glfw should be doing. I've tried to first make my game pixel density agnostic, but the differences between the default behaviors for mac and windows made it impossible without granular access to glfw

**Comment by emidoots** [comment]
@menduz I am going to close this PR. To land this change I need:

1. The `setResizable` change to be picked out and sent as it's own PR, I am happy with this change. Filed https://github.com/hexops/mach/issues/1101 to track this.
2. The `Dockerfile` and other `.devcontainer` files to be removed (I know these were added by accident, no worries)
3. The pixel / display density change needs to be an issue/discussion, I've filed https://github.com/hexops/mach/issues/1102 to kickstart this. I am not convinced that the changes in this PR are correct

I hope that direction sounds okay to you, if any of these changes are blocking you or you want to get them landed sooner let me know and I can prioritize it. Otherwise I can get around to them whenever

**Pull request closed without merging**